### ISSUE TITLE
findComponentRoot unable to find element

### ISSUE THREAD

**Issue by jedwards1211** [intljusticemission/react-big-calendar#394]
I'm getting the following error:

> Uncaught Error: `findComponentRoot(..., .0.0.2.0.0.$1/=11=2$0.$0.0.1.0.1.1:$0.0.0.$header_0.0)`: Unable to find element. This probably means the DOM was unexpectedly mutated (e.g., by the browser), usually due to forgetting a `<tbody>` when using tables, nesting tags like `<form>, <p>, or <a>`, or using non-SVG elements in an `<svg>` parent. Try inspecting the child nodes of the element with React ID ``.

Note, I'm using React 0.14.9.

When I go in the debugger and look at the React node cache, I find that `.0.0.2.0.0.$1/=11=2$0.$0.0.1.0.1.1:$0.0` is the first week row shown in `BigCalendar`:

![screen shot 2017-05-15 at 10 28 18 am](https://cloud.githubusercontent.com/assets/1448194/26065409/789bd798-3959-11e7-9741-b76240f2ff56.png)

Is something funky going on in `EventRowMixin.getRowHeight()`?  (Which I'm surprised doesn't `return` the height?)

**Comment by jedwards1211** [comment]
I thought mutating even a DOM element's style directly is discouraged?
https://github.com/intljusticemission/react-big-calendar/blob/master/src/TimeGrid.js#L408

**Comment by jquense** [comment]
It's "discouraged" because it's not very react-y but in this case it's appropriate. It's probably not the source of your error as it' is supported by React. More likely you have another error above that, that is the "true" error.

**Comment by jedwards1211** [comment]
Okay, any idea where I can hunt for the cause of this error?  I'm creating a very simple calendar:

```js
<BigCalendar
          events={[]}
          date={date}
          view={view}
          onNavigate={onNavigate}
          onView={onView}
        />
```

`onView` and `onNavigate` just put the view/date in URL query params, and I parse and pass those back into the `BigCalendar`.

**Comment by jedwards1211** [comment]
I'm also strangely seeing events flash in and out on Sundays when I navigate between months.  I think that must be related.

**Comment by jedwards1211** [comment]
Well, something is going horribly wrong in other views too but React sure doesn't make it easy to figure out.  I thought maybe it was a duplicate React issue but I can't tell.  Anyway, I'll stop bothering you about this issue.

**Comment by jedwards1211** [comment]
Okay, I figured out what mistake I made, and I'm just posting about it here in case you're wondering. 
 
After a lot of digging, I realized a component I wrote a long time ago was calling `setState` during its `componentWillUpdate`, which I didn't realize at the time is a no-no.  This was triggering a second call to `Month`'s `componentWillReceiveProps` before its `componentWillUpdate`, leading to all kinds of nastiness.

There's an old open issue about what error `setState` should throw during `componentWillUpdate`: https://github.com/facebook/react/issues/2682.
